File: src/ex3/src/my_publisher.cpp
#include <iostream>
#include <ros/ros.h>
#include <std_msgs/Float32.h>
#include <math.h>

using namespace std;

int main(int argc, char* argv[]) {
	
	ros::init(argc, argv, "my_publisher");  
	ros::NodeHandle nh;  
	ros::Publisher pub = nh.advertise<std_msgs::Float32>("sinus", 10);  
	ros::Rate rate(10);
	
	while (ros::ok()) {
		for (int i = 0; i < 100; i++) {
			
			std_msgs::Float32 msg;    
			msg.data = sin(0.01 * i);
			   
			ROS_INFO("Sending: %f", msg.data);    
			pub.publish(msg);   
			ros::spinOnce();    
			rate.sleep();  
		}
	}  
	return 0;
}

File: src/ex2/src/my_publisher.py
#!/usr/bin/python3
import rospy 
from std_msgs.msg import Int32, String


def main(): 
    rospy.init_node('my_publisher')

    pub = rospy.Publisher('text', String, queue_size=10)
    text = 'sth'
    number = 0
    rate = rospy.Rate(2)  # twice per second

    while not rospy.is_shutdown():

        rospy.loginfo("Sending text: " + text + str(number))
        msg = String()
        msg.data = text + str(number)
        pub.publish(msg)

        number += 1
        rate.sleep()

 
if __name__ == '__main__': 
  try: 
      main()
  except rospy.ROSInterruptException: 
      pass
